File: Semestr3/Paradygmaty-Programowania/Cwiczenia/Cwiczenia1/cw2.ml


let rec count (elem , list) =
  if list = [] then 0
  else if List.hd list = elem then 1 + count (elem, List.tl list)
  else count (elem, List.tl list);;


  let value = count (3, [2;2;4;3;3]);;
  print_int value;;

File: Semestr3/Paradygmaty-Programowania/Cwiczenia/Cwiczenia1/cw2.scala
object cw2 extends App {

def count[A](elem: A, list: List[A]): Int = {
  if list == Nil then 0 
  else if list.head == elem then 1 + count(elem, list.tail)
  else count(elem, list.tail)
}

val c = count(3, List(3,4,5,3,2) )
print(c)
    
}